src/hex/shape.rs

```diff
@@ -51,16 +73,176 @@ impl Shape<Axial> {
         }
     }
 
+    pub fn resize(&mut self, scale: (f64, f64)) -> &Self {
+        todo!()
+    }
+
+    // Bake the transform into localspace
+    pub fn bake_transform(&mut self) -> HexShape<T> {
+        todo!()
+    }
+
     // Retrieves the hex array in local space
-    pub fn get_hexes(&self) -> Vec<Axial> {
-        self.tiles.clone()
+    pub fn get_hexes(&self) -> &[HexTile<T>] {
+        self.tiles.as_slice()
     }
 
     // Retrieves the hex array in grid space
-    pub fn get_hexes_gs(&self) -> Vec<Axial> {
+    // Because we modify the HexTile coordinates into gs we must
+    // return a cloned vector instead of a slice.
+    pub fn get_hexes_gs(&mut self) -> Vec<HexTile<T>>
+    where
+        T: Clone
+    {
         self.tiles
             .iter()
-            .map(|&coord| coord.apply_all(self.transform))
+            .map(|tile| {tile.coord.apply_all(self.transform); tile.clone()})
```

Comment:
can this be chained to `tile.coord.apply_all(self.transform).clone()`?

Reply:
This function does not exist anymore. So, no need to refactor it.